packages/flutter/lib/src/rendering/binding.dart

```diff
@@ -200,38 +203,163 @@ mixin RendererBinding on BindingBase, ServicesBinding, SchedulerBinding, Gesture
+  /// [PipelineOwner] and therefore isn't considered during frame production.
+  /// It is also not registered with this binding via [addRenderView].
+  /// Wiring this up is left as a responsibility for a higher level. The
+  /// [WidgetsBinding], for example, sets this up in
+  /// [WidgetsBinding.wrapWithDefaultView], which is called indirectly from
+  /// [runApp].
+  ///
+  /// Apps, that don't use the [WidgetsBinding] or don't call [runApp] (or
+  /// [WidgetsBinding.wrapWithDefaultView]) must manually assign a
+  /// [PipelineOwner] to this [RenderView], make sure the pipeline owner is part
+  /// of the pipeline owner tree rooted at [rootPipelineOwner], and call
+  /// [addRenderView] if they want to use this deprecated property.
+  ///
+  /// Instead of interacting with this deprecated property, consider using
+  /// [renderViews] instead, which contains all [RenderView]s managed by the
+  /// binding.
+  @Deprecated(
+    'Consider using RendererBinding.renderViews instead as the binding may manage multiple RenderViews. '
+    'This feature was deprecated after v3.10.0-12.0.pre.'
+  )
+  // TODO(goderbauer): When this deprecated property is removed also delete the _ReusableRenderView class.
+  late final RenderView renderView = _ReusableRenderView(
+    view: platformDispatcher.implicitView!,
+  );
+
+  /// Creates the [PipelineOwner] that serves as the root of the pipeline owner
+  /// tree ([rootPipelineOwner]).
+  ///
+  /// {@template flutter.rendering.createRootPipelineOwner}
+  /// By default, the root pipeline owner is not setup to manage its own
+  /// [RenderView] and in this configuration nothing must be assigned to
+  /// [PipelineOwner.rootNode] of [rootPipelineOwner]. If necessary,
+  /// [createRootPipelineOwner] may be overridden to create a root pipeline
+  /// owner configured to manage its own [RenderView].
+  ///
+  /// In typical use, children are added to the root pipeline owner (via
+  /// [PipelineOwner.adoptChild]). Those children typically do each manage their
+  /// own [RenderView] and produce distinct render trees which render their
+  /// content into the [FlutterView] associated with that [RenderView].
+  /// {@endtemplate}
+  PipelineOwner createRootPipelineOwner() {
+    return PipelineOwner(onSemanticsUpdate: (ui.SemanticsUpdate update) {
+      assert(() {
+        throw FlutterError.fromParts(<DiagnosticsNode>[
+          ErrorSummary(
+            'The global pipeline owner produced an unexpected semantics update.',
+          ),
+          ErrorDescription(
+            'By default, the RendererBinding.rootPipelineOwner is not configured '
+            'to handle semantics because it is not expected to own a root node.',
+          ),
+          ErrorHint(
+            'Override RendererBinding.createRootPipelineOwner to create a '
+            'pipeline owner that is configured for semantics.',
+          ),
+        ]);
+      }());
```

Comment:
(Capturing my conversation with Michael)

From my understanding, this error is necessary as the user can set the root pipeline owner's `rootNode`, which is unexpected unless you also create a custom root pipeline owner.

Instead, an alternative may be to disallow setting the "normal" root pipeline owner's `rootNode`. This may have the benefit of catching the user error sooner, however, it may have the drawback of added complexity.

Reply:
Finally got around to implement this change. I think its a good one. As you say, it produces the error earlier and hopefully makes it easier to correct.